### PR TITLE
fix: remove pubnub package and associated sync with extension code

### PR DESCRIPTION
**Pull request by sethkfman**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Remove pubnub dependency all legacy source code associated with deprecated sync with extension feature. Should not impact existing application functionality.

**Screenshots/Recordings**

NA

**Issue**

Progresses #???

**Checklist**

* [NA] There is a related GitHub issue
* [NA] Tests are included if applicable
* [NA] Any added code is fully documented

### CODE REVIEW

**Comment by sethkfman** [comment]
@cortisiko @chrisleewilcox I have added spot check on release. This feature is just removing old code that is no longer accessed.

File: app/components/Views/Onboarding/index.js [hunk 1]
```diff
@@ -312,36 +308,6 @@ class Onboarding extends PureComponent {
...
-      );
-      return false;
-    }
-    const action = async () => {
-      const metricsOptIn = await DefaultPreference.get(METRICS_OPT_IN);
-      if (metricsOptIn) {
-        this.props.navigation.navigate('ExtensionSync', {
-          [PREVIOUS_SCREEN]: ONBOARDING,
-        });
-        this.track(MetaMetricsEvents.WALLET_SYNC_STARTED);
```
Comment (Cal-L) [inline app/components/Views/Onboarding/index.js:330]: Remove from metrics

File: app/components/Views/Onboarding/index.js [hunk 2]
```diff
@@ -434,19 +400,6 @@ class Onboarding extends PureComponent {
...
-          {/* Temporarily Disable Sync until the new improved version is ready for release */}
-          {__DEV__ && (
-            <View style={styles.buttonWrapper}>
-              <StyledButton
-                style={styles.button}
-                type={'normal'}
-                onPress={this.onPressSync}
-                testID={'onboarding-import-button'}
-              >
-                {strings('import_wallet.sync_from_browser_extension_button')}
```
Comment (Cal-L) [inline app/components/Views/Onboarding/index.js:446]: Remove from translations files

**Review by Cal-L — COMMENTED** [review]
Lgtm!

**Review by Gudahtt — APPROVED** [review]
LGTM! I didn't do any manual testing but I did review the code changes

**Review by Cal-L — APPROVED** [review]
lgtm!

**Pull request merged**